runtime/src/bank.rs

```diff
@@ -1499,6 +1500,19 @@ impl Bank {
             .clone()
     }
 
+    pub fn new_program_cache_for_tx_batch_for_slot(&self, slot: Slot) -> LoadedProgramsForTxBatch {
+        let loaded_program_cache_r = self.loaded_programs_cache.read().unwrap();
+        let epoch = self.epoch_schedule.get_epoch(slot);
+        LoadedProgramsForTxBatch::new(
```

Comment:
Think this should be moved into a `LoadedProgramsForTxBatch` constructor and not happen in the bank.

Reply:
Moved most of the code to its constructor. Also, moved `new_program_cache_for_tx_batch_for_slot()` to `dev-context-only-util` since it's only used by ledger-tool. This function is needed, as it's accessing non-public `bank` fields.